### PR TITLE
CDAP-6783 Fixed the classpath ordering in Spark to load the classes from the cdap-common first.

### PR DESCRIPTION
**Pull request by sagarkapare**
JIRA: https://issues.cask.co/browse/CDAP-6783

### CODE REVIEW

**Comment by albertshau** [comment]
lgtm. was tested on a cluster.

**Comment by gokulavasan** [comment]
@sagarkapare would it be possible to add some comments?

File: cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkRuntimeService.java [hunk 1]
```diff
@@ -362,6 +364,19 @@ private File createLauncherJar(File tempDir) throws IOException {
     return jarFile;
   }
 
+  @Nullable
+  private static String getCdapCommonJarName() {
+    try {
+      CodeSource codeSource = CConfiguration.class.getProtectionDomain().getCodeSource();
+      if (codeSource != null) {
```
Comment (anwar6953) [inline cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkRuntimeService.java:371]: what happens if codeSource is null?
Should we log a warning in that case also?

Comment (sagarkapare) [inline cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkRuntimeService.java:371]: Added warn log in caller for this method.

**Comment by sagarkapare** [comment]
@anwar6953 @gokulavasan added comments. please take a look again. thanks!

**Comment by anwar6953** [comment]
lgtm

**Comment by sagarkapare** [comment]
Build green here - https://builds.cask.co/browse/CDAP-DUT4548-2. Merging.

**Pull request merged**